### PR TITLE
Reduce server error logs logged for each connection

### PR DESCRIPTION
**Pull request by rastislavs**
This PR resolves the server part of the issue mentioned in https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/319

During normal server operation,  the following errors are logged pretty-much for every closed connection:
```
E0617 13:46:51.708548       1 server.go:390] "Stream read from frontend failure" err="rpc error: code = Canceled desc = context canceled"                                                                                                                                                                                       
E0617 13:46:51.710667       1 server.go:761] "could not get frontend client" err="can't find connID 4 in the frontends[0abaddef-7be4-469c-9a40-ee8e38d7d981]" serverID="04ac2b6d-bde4-4680-80df-1e0d87d34bb1" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=4
```

That makes the production operation hard, as it is not easy to spot / alert on real errors then.

For the `context canceled` error I could not really find the root cause, so I just added a specific check for cancelled error and lifted it to an info log.

Regarding the `could not get frontend client` error on `CLOSE_RSP`, it seems that we are getting multiple (or always 2) `CLOSE_RSP` for each closed connection, and the error is triggered upon receiving the second one. Not sure if it was intended to receive multiple `CLOSE_RSP` and how much of an issue it is, but since it seems to be the normal operational flow, I lifted the error to an info log.

Logs confirming 2 CLOSE_RSP for each connection ID:
```                                                           
I0617 14:06:57.379055       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=1
I0617 14:06:57.379089       1 server.go:768] CLOSE_RSP sent to frontend connectionID 1
I0617 14:06:57.383220       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=1
I0617 14:06:57.383576       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=4
I0617 14:06:57.383639       1 server.go:768] CLOSE_RSP sent to frontend connectionID 4
I0617 14:06:57.383688       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=3
I0617 14:06:57.383706       1 server.go:768] CLOSE_RSP sent to frontend connectionID 3
I0617 14:06:57.384982       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="1649efbc-492f-4e57-8e17-68ec850961ab" connectionID=2
I0617 14:06:57.385010       1 server.go:768] CLOSE_RSP sent to frontend connectionID 2
I0617 14:06:57.386464       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=4
I0617 14:06:57.386507       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="0abaddef-7be4-469c-9a40-ee8e38d7d981" connectionID=3
I0617 14:06:57.388036       1 server.go:758] "Received CLOSE_RSP" serverID="35e2ed92-3605-4ae0-b454-50974d97dcee" agentID="1649efbc-492f-4e57-8e17-68ec850961ab" connectionID=2
```

Signed-off-by: Rastislav Szabo <rastislav@kubermatic.com>

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -386,8 +388,14 @@ func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
 				close(stopCh)
 				return
 			}
+			if status.Code(err) == codes.Canceled {
+				klog.V(2).InfoS("Stream read from frontend cancelled", "userAgent", userAgent, "serverID", s.serverID)
```
Comment (rata) [inline pkg/server/server.go:392]: Do we want to move this inside the `err != nil` case below? We would avoid the duplicated code to send the error and close the channel

**Comment by rata** [comment]
You can amend the commit instead of creating a new one to address my review comment :)

File: pkg/server/server.go [hunk 2]
```diff
@@ -766,7 +772,8 @@ func (s *ProxyServer) serveRecvBackend(backend Backend, stream agent.AgentServic
 			klog.V(5).InfoS("Received CLOSE_RSP", "serverID", s.serverID, "agentID", agentID, "connectionID", resp.ConnectID)
 			frontend, err := s.getFrontend(agentID, resp.ConnectID)
 			if err != nil {
-				klog.ErrorS(err, "could not get frontend client", "serverID", s.serverID, "agentID", agentID, "connectionID", resp.ConnectID)
+				// assuming it is already closed, just log it
+				klog.V(4).InfoS("could not get frontend client for closing", "serverID", s.serverID, "agentID", agentID, "connectionID", resp.ConnectID, "err", err)
```
[the remark this answers is not in the document]
Comment (rastislavs) [reply pkg/server/server.go]: done

**Pull request merged**